### ISSUE TITLE
Updates to hashing structure changes test output

### ISSUE THREAD

**Issue by ehildenb** [runtimeverification/haskell-backend#3393]
Here, we isolated that #3338 causes a change in some test output in the frontend. It really doesn't make sense _why_ this is causing test output change, so we should understand that.

For now, we've updated the expected output, because it appears to be correct.

I would suggest that we first find out _where/when_ the additional side conditions are added to the constraints (by tracing somehow). Then we can look at whether they are introduced on both the old and new versions (before and after #3338 ). If both versions instroduce this, but the prior versions somehow eliminates those side conditions, then we can trace where those side-conditions are being simplified out by the old version (but not the new version). If the old version just never even introduces those side conditions, then we should look at why the new version is.

First step is porting this test to a regression test on teh backend itself.

**Comment by goodlyrottenapple** [comment]
I did a few more experiments and there is something fishy about how we use the Hashable instance in the first place. I went back to one commit before David's changes landed and confirmed that the test output was the same. Then I made the TermLike Hashable instance into a dummy hash, i.e. just returning the salt. This means you essentially always get a hash collision (I think). With this change, I saw the new output for the above test. This seems problematic since the assumption for using the hash in places like a HashMap is that it shouldn't matter how the hash is implemented.

I've added the test on this branch: https://github.com/runtimeverification/haskell-backend/pull/3390

can individually be called via

```
nix-build test.nix --argstr test set/inset-11
```
